test-cases/fac_error.erl:8: Type error: type mismatch;
  found   : 'foo'
  required: number()
there is no solution that satisfies subtype constraints
